### PR TITLE
mesos-slave "node-groups" attribute

### PR DESCRIPTION
**Pull request by unakatsuo**
#158 raised a concern that we will need to maintain more attributes as part of codebase when the scheduler gets more dimensions to calculate VM allocation.

This change generalize the idea to attribute the mesos slave node with the group names.

CLI's optional JSON parameter was changed that the scheduler can allocate the instance to  the slave node having the group names.

```
openvdc run centos/7/lxc '{"interfaces":[{"type":"veth"}], "node_groups":["linuxbr"]}'
```

### CODE REVIEW

**Comment by unakatsuo** [comment]
```
--- FAIL: TestLXCInstance_NICx2 (20.81s)
	02_start_and_terminate_lxc_instance_test.go:56: eth1
	02_start_and_terminate_lxc_instance_test.go:58: Interface i-0000000001_00 is not attached
	02_start_and_terminate_lxc_instance_test.go:65: i-0000000001_00
	02_start_and_terminate_lxc_instance_test.go:67: Interface i-0000000001_01 is not attached
```

**Comment by unakatsuo** [comment]
https://ci.openvdc.org/blue/rest/organizations/jenkins/pipelines/citest/branches/attrib-node-groups/runs/7/nodes/36/log/

```
=== RUN   TestLXCInstance
--- FAIL: TestLXCInstance (351.11s)
	00_run_cmd.go:115: Timed out for 300 sec
	00_ssh.go:77: STDOUT:
	00_ssh.go:78: STDERR:
		i-0000000000 doesn't exist
	00_ssh.go:80: Running 'sudo lxc-info -n i-0000000000' on '10.0.100.14' failed. Tried 10 times.
=== RUN   TestLXCInstance_NICx2
--- FAIL: TestLXCInstance_NICx2 (348.84s)
	00_run_cmd.go:115: Timed out for 300 sec
	00_ssh.go:77: STDOUT:
	00_ssh.go:78: STDERR:
		i-0000000001 doesn't exist
	00_ssh.go:80: Running 'sudo lxc-info -n i-0000000001' on '10.0.100.14' failed. Tried 10 times.
```

**Comment by unakatsuo** [comment]
https://ci.openvdc.org/blue/rest/organizations/jenkins/pipelines/citest/branches/attrib-node-groups/runs/10/nodes/36/log/
```
--- FAIL: TestOpenVDCLog (40.62s)
	00_run_cmd.go:32: STDOUT:
	00_run_cmd.go:33: STDERR:
		time="2017-05-17T05:27:23Z" level=fatal msg="Error streaming log" error="rpc error: code = 2 desc = cl.GetLog: Get http://executor-lxc-ovs:5051/slave(1)/state.json: dial tcp: lookup executor-lxc-ovs: no such host" 
	00_run_cmd.go:35: Unable to run command: 'openvdc [log i-0000000003]'
		exit status 1
```

**Comment by unakatsuo** [comment]
https://ci.openvdc.org/blue/rest/organizations/jenkins/pipelines/citest/branches/attrib-node-groups/runs/12/nodes/36/log

```
=== RUN   TestCmdReboot
--- FAIL: TestCmdReboot (56.17s)
	00_ssh.go:77: STDOUT:
	00_ssh.go:78: STDERR:
		lxc-attach: attach.c: lxc_attach: 752 failed to get the init pid
	00_ssh.go:80: Running 'sudo lxc-attach -n i-0000000004 -- chmod 755 /etc/rc.d/rc.local' on '10.0.100.14' failed. Tried 10 times.
```

```
May 17 16:34:02 executor-lxc sshd[12122]: pam_unix(sshd:session): session opened for user kemumaki by (uid=0)
May 17 16:34:02 executor-lxc sudo[12125]: kemumaki : TTY=unknown ; PWD=/home/kemumaki ; USER=root ; COMMAND=/bin/lxc-attach -n i-0000000004 -- chmod 755 /etc/rc.d/rc.local
May 17 16:34:02 executor-lxc sshd[12122]: pam_unix(sshd:session): session closed for user kemumaki
May 17 16:34:02 executor-lxc systemd-logind[406]: Removed session 18.
May 17 16:34:02 executor-lxc systemd[1]: Removed slice user-1000.slice.
May 17 16:34:02 executor-lxc systemd[1]: Stopping user-1000.slice.
May 17 16:34:07 executor-lxc sshd[12135]: Accepted password for kemumaki from 10.0.100.1 port 47660 ssh2
May 17 16:34:07 executor-lxc systemd[1]: Created slice user-1000.slice.
May 17 16:34:07 executor-lxc systemd[1]: Starting user-1000.slice.
May 17 16:34:07 executor-lxc systemd[1]: Started Session 19 of user kemumaki.
May 17 16:34:07 executor-lxc systemd-logind[406]: New session 19 of user kemumaki.
May 17 16:34:07 executor-lxc systemd[1]: Starting Session 19 of user kemumaki.
May 17 16:34:07 executor-lxc sshd[12135]: pam_unix(sshd:session): session opened for user kemumaki by (uid=0)
May 17 16:34:07 executor-lxc sudo[12138]: kemumaki : TTY=unknown ; PWD=/home/kemumaki ; USER=root ; COMMAND=/bin/lxc-attach -n i-0000000004 -- chmod 755 /etc/rc.d/rc.local
May 17 16:34:07 executor-lxc sshd[12135]: pam_unix(sshd:session): session closed for user kemumaki
May 17 16:34:07 executor-lxc systemd-logind[406]: Removed session 19.
May 17 16:34:07 executor-lxc systemd[1]: Removed slice user-1000.slice.
May 17 16:34:07 executor-lxc systemd[1]: Stopping user-1000.slice.
May 17 16:34:12 executor-lxc sshd[12148]: Accepted password for kemumaki from 10.0.100.1 port 47662 ssh2
May 17 16:34:12 executor-lxc systemd[1]: Created slice user-1000.slice.
May 17 16:34:12 executor-lxc systemd[1]: Starting user-1000.slice.
May 17 16:34:12 executor-lxc systemd[1]: Started Session 20 of user kemumaki.
May 17 16:34:12 executor-lxc systemd-logind[406]: New session 20 of user kemumaki.
May 17 16:34:12 executor-lxc sshd[12148]: pam_unix(sshd:session): session opened for user kemumaki by (uid=0)
May 17 16:34:12 executor-lxc systemd[1]: Starting Session 20 of user kemumaki.
May 17 16:34:12 executor-lxc sudo[12151]: kemumaki : TTY=unknown ; PWD=/home/kemumaki ; USER=root ; COMMAND=/bin/lxc-attach -n i-0000000004 -- chmod 755 /etc/rc.d/rc.local
May 17 16:34:12 executor-lxc sshd[12148]: pam_unix(sshd:session): session closed for user kemumaki
```

```
May 17 16:33:22 executor-lxc-ovs mesos-slave[1041]: I0517 16:33:22.625267  1242 slave.cpp:1625] Got assigned task 'i-0000000004' for framework scheduler_1
May 17 16:33:22 executor-lxc-ovs mesos-slave[1041]: I0517 16:33:22.625419  1242 slave.cpp:1785] Launching task 'i-0000000004' for framework scheduler_1
May 17 16:33:22 executor-lxc-ovs mesos-slave[1041]: I0517 16:33:22.625469  1242 slave.cpp:2140] Queued task 'i-0000000004' for executor 'vdc-hypervisor-lxc' of framework scheduler_1 at executor(1)@10.0.100.15:55718
May 17 16:33:22 executor-lxc-ovs mesos-slave[1041]: I0517 16:33:22.625692  1242 slave.cpp:2331] Sending queued task 'i-0000000004' to executor 'vdc-hypervisor-lxc' of framework scheduler_1 at executor(1)@10.0.100.15:55718
May 17 16:33:22 executor-lxc-ovs vdc-executor[11359]: 2017-05-17 16:33:22 [INFO] openvdc-executor/main.go:61 Launching task VDC_i-0000000004 with command
May 17 16:33:22 executor-lxc-ovs mesos-slave[1041]: I0517 16:33:22.628803  1242 slave.cpp:3816] Handling status update TASK_RUNNING (UUID: 1b28dd9a-3ad3-11e7-a213-525601000506) for task i-0000000004 of framework scheduler_1 from executor(1)@10.0.100.15:55718
May 17 16:33:22 executor-lxc-ovs mesos-slave[1041]: I0517 16:33:22.629230  1242 status_update_manager.cpp:323] Received status update TASK_RUNNING (UUID: 1b28dd9a-3ad3-11e7-a213-525601000506) for task i-0000000004 of framework scheduler_1
May 17 16:33:22 executor-lxc-ovs mesos-slave[1041]: I0517 16:33:22.629425  1242 slave.cpp:4256] Forwarding the update TASK_RUNNING (UUID: 1b28dd9a-3ad3-11e7-a213-525601000506) for task i-0000000004 of framework scheduler_1 to master@10.0.100.11:5050
May 17 16:33:22 executor-lxc-ovs mesos-slave[1041]: I0517 16:33:22.629510  1242 slave.cpp:4166] Sending acknowledgement for status update TASK_RUNNING (UUID: 1b28dd9a-3ad3-11e7-a213-525601000506) for task i-0000000004 of framework scheduler_1 to executor(1)@10.0.100.15:55718
May 17 16:33:22 executor-lxc-ovs mesos-slave[1041]: I0517 16:33:22.633345  1242 status_update_manager.cpp:395] Received status update acknowledgement (UUID: 1b28dd9a-3ad3-11e7-a213-525601000506) for task i-0000000004 of framework scheduler_1
May 17 16:33:22 executor-lxc-ovs vdc-executor[11359]: 2017-05-17 16:33:22 [INFO] openvdc-executor/main.go:127 Creating instance hypervisor=lxc instance_id=i-0000000004 state=STARTING
May 17 16:33:22 executor-lxc-ovs vdc-executor[11359]: 2017-05-17 16:33:22 [INFO] github.com/axsh/openvdc/hypervisor/lxc/lxc.go:248 Creating lxc-container... hypervisor=lxc instance_id=i-0000000004
May 17 16:33:31 executor-lxc-ovs vdc-executor[11359]: 2017-05-17 16:33:31 [DEBUG] github.com/axsh/openvdc/hypervisor/lxc/lxc.go:175 resource:lxc_template:<template:"download" distro:"centos" release:"7" >  hypervisor=lxc instance_id=i-0000000004
May 17 16:33:31 executor-lxc-ovs vdc-executor[11359]: 2017-05-17 16:33:31 [DEBUG] github.com/axsh/openvdc/hypervisor/lxc/lxc.go:140 # Template used to create this container: /usr/share/lxc/templates/lxc-download
                                                      # Parameters passed to the template: --dist centos --release 7 --arch amd64
                                                      # For additional config options, please look at lxc.container.conf(5)
                                                      
                                                      # Distribution configuration
                                                      lxc.include = /usr/share/lxc/config/centos.common.conf
                                                      lxc.arch = x86_64
                                                      
                                                      # Container specific configuration
                                                      lxc.rootfs = /var/lib/lxc/i-0000000004/rootfs
                                                      lxc.utsname = i-0000000004
                                                      
                                                      # Network configuration
                                                      lxc.network.type = veth
                                                      lxc.network.flags = up
                                                      lxc.network.link = virbr0
                                                      
                                                      # OpenVDC Network Configuration
                                                      # Here clear all network options.
                                                      lxc.network.type=
                                                       hypervisor=lxc instance_id=i-0000000004
May 17 16:33:31 executor-lxc-ovs vdc-executor[11359]: 2017-05-17 16:33:31 [INFO] openvdc-executor/main.go:133 Starting instance hypervisor=lxc instance_id=i-0000000004 state=STARTING
May 17 16:33:31 executor-lxc-ovs vdc-executor[11359]: 2017-05-17 16:33:31 [INFO] github.com/axsh/openvdc/hypervisor/lxc/lxc.go:326 Starting lxc-container... hypervisor=lxc instance_id=i-0000000004
May 17 16:33:31 executor-lxc-ovs vdc-executor[11359]: 2017-05-17 16:33:31 [INFO] github.com/axsh/openvdc/hypervisor/lxc/lxc.go:331 Waiting for lxc-container to become RUNNING hypervisor=lxc instance_id=i-0000000004
May 17 16:33:31 executor-lxc-ovs vdc-executor[11359]: 2017-05-17 16:33:31 [INFO] openvdc-executor/main.go:140 Instance launched successfully hypervisor=lxc instance_id=i-0000000004 state=STARTING
```

File: ci/citest/acceptance-test/multibox/10.0.100.14-vdc-executor-lxc/guestroot/etc/mesos-slave/attributes [hunk 1]
```diff
@@ -1 +1 @@
-hypervisor:lxc
+hypervisor:lxc;node-groups:linuxbr
```
Comment (Metallion) [inline ci/citest/acceptance-test/multibox/10.0.100.14-vdc-executor-lxc/guestroot/etc/mesos-slave/attributes:1]: Is it possible to have this on two lines?

```
hypervisor:lxc
node-groups:linuxbr
```

Also, is it possible to have multiple groups comma separated?

```
node-groups: linuxbr, foobar, hoge
```

Comment (unakatsuo) [reply ci/citest/acceptance-test/multibox/10.0.100.14-vdc-executor-lxc/guestroot/etc/mesos-slave/attributes:1]: I don't think it's possible for multiline: See /usr/bin/mesos-init-wrapper.

For comma, scheduler/mesos.go:134

**Comment by unakatsuo** [comment]
https://ci.openvdc.org/blue/rest/organizations/jenkins/pipelines/citest/branches/attrib-node-groups/runs/16/nodes/36/log/?start=0

```
** DOING STEP: Mount temporary root folder for zookeeper
losetup: /multibox/10.0.100.10-zookeeper/box-disk1.raw: failed to set up loop device: No such file or directory
Error occured with losetup command (rc=1) or output was unexpected ().
Script failed...exiting. ()
** Skipping step: Unmount temporary root folder for zookeeper
Script failed...exiting. ()
+ cleanup
```

**Review by b0r6 — APPROVED** [review]

**Pull request merged**